packages/editor/src/components/rich-text/index.js

```diff
@@ -874,7 +875,8 @@ export class RichText extends Component {
 								tagName={ Tagname }
 								onSetup={ this.onSetup }
 								style={ style }
-								defaultValue={ this.valueToEditableHTML( record ) }
+								record={ record }
```

Comment:
mmm, wouldn't this have the same effect than before?

This is what I see:

- the `record` value is created on every `RichText` render by the `getRecord` function.
- the `getRecord` function returns a new object every time is called.
- on every `RichText` render, the `TinyMCE` component will receive a new object reference for the `record` prop (although values may be the same, I haven't looked at what this function does), so it'll be re-rendered.

Reply:
oh, I just saw that the `TinyMCE` defines a `shouldComponentUpdate` that always returns `false`. So what this PR is trying to avoid is to actually call `valueToEditableHTML` in every `RichText` render, not re-renders of the child component.

Reply:
Right, `valueToEditableHTML` will only be called on mount there.

Reply:
So, my understanding is that this would reduce the time it takes RichText to execute in every render/commit phase. I've tested this by:

* Opened the Gutenberg demo.
* In the first paragraph introduced the text "Howdy. ".

These are the results in `master`:

![peek 2018-11-30 14-03-master-rich-text](https://user-images.githubusercontent.com/583546/49291206-cb2a5680-f4a9-11e8-8e87-129f77c89ebd.gif)

These are the results in this branch:

![peek 2018-11-30 13-50-branch-rich-text](https://user-images.githubusercontent.com/583546/49291212-d4b3be80-f4a9-11e8-8585-c1e3ae635e36.gif)

Perhaps is difficult to observe in the GIFs, but it looks like RichText takes ~0,5ms to render in every keystroke in both `master` and this branch. I don't see at the moment which other tests I could do to effectively measure the reduction of the performance budget introduced by this branch. 

Nevertheless, doing less work is always better! So, I think it's fine to merge this.

Reply:
I'm not sure how to interpret this graph. What is it showing?

Reply:
A new record being created? Which function exactly?

Reply:
I'm from mobile now.

This is from React Developer Tools new profiler (to be found in the React tab). You can navigate every commit and components that were updated will be colored (blueish to yellow, yellow the ones that took the most time).

The `getRecord` method in RichText does something like:

```
getRecord(){
    ...
    return {
       something,
       otherSomething,
    };
}
```

essentially returning a new object no matter whether `something` values have been updated or not.

Reply:
Where does the result of `getRecord` get compared?

Reply:
Oh, I see now. Good catch.

Reply:
My understanding is that React does a shallow comparison of the props passed to the components to decide whether a particular component needs to be re-rendered. We have two scape-hatches for that: 1) always pass the same prop (in the case of arrays and objects, the same reference) or 2) hook into `shouldComponentUpdate` to do that process ourselves.